services/ad-tech/src/routes/seller-router.ts

```diff
@@ -0,0 +1,269 @@
+  // Collect buyer signals from contextual bids.
+  const buyerSignals: {[key: string]: any} = {};
+  for (const contextualBid of contextualBids) {
+    if (contextualBid.buyerSignals) {
+      buyerSignals[contextualBid.bidder!] = contextualBid.buyerSignals;
+    }
+  }
+  const response = {
+    bidder: CURRENT_ORIGIN,
+    auctionId: signals['auctionId'],
+    bid: winningContextualBid.bid,
+    renderURL: winningContextualBid.renderURL,
+    componentAuctionConfig: constructAuctionConfig({
+      useCase: req.query.useCase?.toString(),
+      adType: req.query.adType?.toString(),
+      auctionSignals: signals,
+      buyerSignals,
+    }),
+  };
+  console.log('Responding to contextual bid request', {response});
+  res.json(response);
+});
+
+/** Returns the PAAPI auction config. */
+SellerRouter.get(
+  '/auction-config.json',
+  async (req: Request, res: Response) => {
+    const signals: {[key: string]: string} = {};
+    for (const key of Object.keys(req.query)) {
+      signals[key] = req.query[key]?.toString() || '';
+    }
+    const auctionConfig = constructAuctionConfig({
+      useCase: req.query.useCase?.toString(),
+      adType: req.query.adType?.toString(),
+      auctionSignals: signals,
+    });
+    console.log('Returning auction config: ', {auctionConfig});
+    res.json(auctionConfig);
+  },
+);
+
+/** Simplified BYOS implementation for Key-Value Service. */
+SellerRouter.get(
+  '/scoring-signal.json',
+  async (req: Request, res: Response) => {
+    // TODO: Partition KV keys by publisher.
+    const publisher = req.query.hostname;
+    const queryRenderUrls = req.query.renderUrls?.toString().split(',') || [];
+    const renderUrlMetadata =
+      trustedScoringSignalsStore.getMultiple(queryRenderUrls);
+    const queryAdAcomponentRenderUrls =
+      req.query.adComponentRenderUrls?.toString().split(',') || [];
+    const adComponentRenderUrlMetadata = trustedScoringSignalsStore.getMultiple(
+      queryAdAcomponentRenderUrls,
+    );
+    console.log(
+      'KV BYOS querying',
+      {queryRenderUrls, queryAdAcomponentRenderUrls},
+      'for',
+      {publisher},
+    );
+    const scoringSignals = {
+      renderURLs: {...renderUrlMetadata},
+      adComponentRenderURLs: {...adComponentRenderUrlMetadata},
+    };
+    res.setHeader('X-Allow-FLEDGE', 'true');
+    // res.setHeader('X-fledge-bidding-signals-format-version', '2');
+    console.log('Returning trusted scoring signals: ', {scoringSignals});
+    res.json(scoringSignals);
+  },
+);
+
+/** Adds values in query parameters to Key Value Store. */
+SellerRouter.get(
+  '/set-scoring-signal.json',
+  async (req: Request, res: Response) => {
+    console.log('Setting scoring signals', {...req.query});
+    for (const key of Object.keys(req.query)) {
+      trustedScoringSignalsStore.set(key, req.query[key]?.toString());
+    }
+    res.sendStatus(200);
+  },
+);
+
+/** Rewrites the default values in the key value store. */
+SellerRouter.get(
```

Comment:
These utility actions are dangerous depending on their exposure/expectation of usage. Maybe move it down to an npm run task through docker.